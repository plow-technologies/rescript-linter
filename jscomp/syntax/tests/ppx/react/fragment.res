@@jsxConfig({version:4, mode: "classic"})

let _ = <></>
let _ = <><div /></>
let _ = <><div /><div /></>
let _ = <><></></>
let _ = <Z />
let _ = <Z><div /></Z>
let _ = <Z a="a"><div /></Z>
let _ = <Z><div /><div /></Z>
let _ = <div />
let _ = <div><div /></div>
let _ = <div id="id"><div /></div>
let _ = <div><div /><div /></div>

@@jsxConfig({version:4, mode: "automatic"})

let _ = <></>
let _ = <><div /></>
let _ = <><div /><div /></>
let _ = <><></></>
let _ = <Z />
let _ = <Z><div /></Z>
let _ = <Z a="a"><div /></Z>
let _ = <Z><div /><div /></Z>
let _ = <div />
let _ = <div><div /></div>
let _ = <div id="id"><div /></div>
let _ = <div><div /><div /></div>