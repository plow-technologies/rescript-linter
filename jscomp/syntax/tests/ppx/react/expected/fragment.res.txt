@@jsxConfig({version: 4, mode: "classic"})

let _ = React.createElement(React.fragment, {})
let _ = React.createElement(
  React.fragment,
  {children: ReactDOM.createDOMElementVariadic("div", [])},
)
let _ = React.createElementVariadic(
  React.fragment,
  {},
  [ReactDOM.createDOMElementVariadic("div", []), ReactDOM.createDOMElementVariadic("div", [])],
)
let _ = React.createElement(React.fragment, {children: React.createElement(React.fragment, {})})
let _ = React.createElement(Z.make, {})
let _ = React.createElement(Z.make, {children: ReactDOM.createDOMElementVariadic("div", [])})
let _ = React.createElement(
  Z.make,
  {a: "a", children: ReactDOM.createDOMElementVariadic("div", [])},
)
let _ = React.createElementVariadic(
  Z.make,
  {children: React.null},
  [ReactDOM.createDOMElementVariadic("div", []), ReactDOM.createDOMElementVariadic("div", [])],
)
let _ = ReactDOM.createDOMElementVariadic("div", [])
let _ = ReactDOM.createDOMElementVariadic("div", [ReactDOM.createDOMElementVariadic("div", [])])
let _ = ReactDOM.createDOMElementVariadic(
  "div",
  ~props={id: "id"},
  [ReactDOM.createDOMElementVariadic("div", [])],
)
let _ = ReactDOM.createDOMElementVariadic(
  "div",
  [ReactDOM.createDOMElementVariadic("div", []), ReactDOM.createDOMElementVariadic("div", [])],
)

@@jsxConfig({version: 4, mode: "automatic"})

let _ = React.jsx(React.jsxFragment, {})
let _ = React.jsx(React.jsxFragment, {children: ReactDOM.jsx("div", {})})
let _ = React.jsxs(
  React.jsxFragment,
  {children: React.array([ReactDOM.jsx("div", {}), ReactDOM.jsx("div", {})])},
)
let _ = React.jsx(React.jsxFragment, {children: React.jsx(React.jsxFragment, {})})
let _ = React.jsx(Z.make, {})
let _ = React.jsx(Z.make, {children: ReactDOM.jsx("div", {})})
let _ = React.jsx(Z.make, {a: "a", children: ReactDOM.jsx("div", {})})
let _ = React.jsxs(
  Z.make,
  {children: React.array([ReactDOM.jsx("div", {}), ReactDOM.jsx("div", {})])},
)
let _ = ReactDOM.jsx("div", {})
let _ = ReactDOM.jsx("div", {children: ?ReactDOM.someElement(ReactDOM.jsx("div", {}))})
let _ = ReactDOM.jsx("div", {id: "id", children: ?ReactDOM.someElement(ReactDOM.jsx("div", {}))})
let _ = ReactDOM.jsxs(
  "div",
  {children: React.array([ReactDOM.jsx("div", {}), ReactDOM.jsx("div", {})])},
)
